kdump/setup-bare-metal/runtest.sh

```diff
@@ -0,0 +1,196 @@
+        elif [[ "${K_ARCH}"  = "x86_64" ]]; then args="crashkernel=128M@16M"
+        elif [[ "${K_ARCH}"  = "ppc64"  ]]; then args="crashkernel=256M@32M xmon=off"
+        elif [[ "${K_ARCH}"  = "ia64"   ]]; then args="crashkernel=512M@256M";
+            # the larger IA-64 box, the more kdump memory needed
+            grep -qE '^ACPI:.*(rx8640|SGI)' /var/log/dmesg &&
+            args="crashkernel=768M@256M"
+        fi
+    fi
+
+    echo "$args"
+}
+
+
+SetupKdump()
+{
+    if [ ! -f "${K_REBOOT}" ]; then
+        rpm -q lshw || yum install -y lshw
+
+        local default=/boot/vmlinuz-`uname -r`
+        [ ! -s "$default" ] && default=/boot/vmlinux-`uname -r`
+
+        # Temporarily comment out this line. Because it seems
+        # if it's executed before rebuilding kdump img,
+        # it may break the grub file and system would hange
+        # after rebooting.
+        #/sbin/grubby --set-default="${default}"
+
+        # In ia64 arch, the path of vmlinuz is /boot/efi/efi/redhat, it different with other arch.
+        if [[ "${K_ARCH}"  = "ia64" ]]; then
+            default=/boot/efi/efi/redhat/vmlinuz-`uname -r`
+            /sbin/grubby --set-default="${default}"
+        fi
+
+        # For uncompressed kernel, i.e. vmlinux
+        [[ ${default} == *vmlinux* ]] && {
+            Log "- Modifying /etc/sysconfig/kdump properly for 'vmlinux'."
+            sed -i 's/\(KDUMP_IMG\)=.*/\1="vmlinux"/' /etc/sysconfig/kdump
+        }
+
+        # For kernel-rt
+        $IS_RT_KEN && [ -f /usr/bin/rt-setup-kdump ] && {
+            Log "- Modifying /etc/sysconfig/kdump properly for RT."
+            set -x; /usr/bin/rt-setup-kdump -g; set +x
+        }
+
+        # Ensure Kdump Kernel memory reservation
+        grep -q 'crashkernel' <<< "${KER1ARGS}" || {
+            local kdumpMem=$(DefKdumpMem)
+            [ -z "${KER1ARGS}" ] || kdumpMem=" ${kdumpMem}"
+
+            $IS_RHEL5 && KER1ARGS+="${kdumpMem}" || {
+                # memory >= auto-threshold and kdump already default on
+                [ `cat /sys/kernel/kexec_crash_size` -eq 0 ] && {
+                    Log "`grep MemTotal /proc/meminfo`"
+                    KER1ARGS+="${kdumpMem}"
+                }
+            }
+        }
+        [ "${KER1ARGS}" ] && {
+            touch "${K_REBOOT}"
+
+            # Kdump service will not be enabled if crashkernel=auto && system
+            # memory is less the threshold required by kdump service.
+            /bin/systemctl enable kdump.service || /sbin/chkconfig kdump on
+
+            Log "- Changing boot loader."
+            {
+                /sbin/grubby                     \
+                    --args="${KER1ARGS}"         \
+                    --update-kernel="${default}" &&
+                if [ "${K_ARCH}" = "s390x" ]; then zipl; fi
+            } || FatalError "Error changing boot loader."
+
+            Report 'pre-reboot'
+            Log "- Rebooting\n"; sync; rhts-reboot
```

Comment:
if KER1ARGS is set in previous steps it would call grubby to update kernel option and manage a reboot.